### DISCUSSION TITLE
Regarding AI-deck and swarming

### DISCUSSION THREAD

**Discussion by Kaijun101** [bitcraze/discussions#400] [Q&A]
Hi Bitcraze!

I have questions regarding using the AI deck for crazyflie. 

So, I  plan to operate a team of 15-20 crazyflies for exploration tasks. So part of the task involves the crazyflies identifying apriltags as targets. Thus, I am lost as to how to implement this. How do I stream back footages for all drones and will it be a bottleneck when using wifi. Also, if streaming back all drone footages for processing on a computer is not possible, would it be possible to run the apriltag detection algo onboard and how so I achieve this?

 Appreciate your insights to these questions I had. 😊

**Comment by evoggy** [comment]
Hi!

It should be possible to stream back the images from all the decks, but there's no software written to receive images from multiple AI decks. Have a look at the [fpv.py](https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/aideck/fpv.py) example where an image is downloaded from one deck.

Unfortunately I cannot say that much about the onboard processing, this is probably something you'll have to test yourself.

**Reply by Kaijun101** [reply]
Hi @evoggy, thanks for your reply! 

I'm actually quite lost in how to deploy code on AI-deck. I saw some neural networks on AI-deck, but don't quite follow the workflow how they are able to run it onboard. 😢 Appreciate if you could provide some assistance on this.

**Reply by Kaijun101** [reply]
Hi Bitcraze, another question I had is how do I set the AIdeck to connect to an access point instead of acting as an access point, as in the image stream example?

**Reply by evoggy** [reply]
Yes, you can set it to connect to an access point instead.

Please have a look at the examples for the AI deck to get a better understanding on how to work with it: https://github.com/bitcraze/aideck-gap8-examples/tree/master/examples

**Reply by Kaijun101** [reply]
Hi @evoggy where can I find information on running self written algo, i.e., apriltags tracking, onboard the AI-Deck, I see most of the online instructions deal with neural networks.

**Reply by evoggy** [reply]
You can also have a look at the Greenwaves SDK examples, they have a bit more than we do: https://github.com/GreenWaves-Technologies/gap_sdk/tree/release-v4.8.0/examples

**Reply by Kaijun101** [reply]
Thanks a lot @evoggy! 

Then, I'm curious if the output of AI-deck can be sent to crazyflie and sent back to some off board computer?
Or if crazyflie can be solely controlled with WiFi?

**Reply by knmcguire** [reply]
Check out this blogpost: https://www.bitcraze.io/2022/09/using-the-ai-deck-for-crtp-over-wifi-via-cpx/. but beware of the current issues we are having with it now, like this one: https://github.com/bitcraze/discussions/discussions/267

For any other questions about this, please start a new discussion topic
